### ISSUE TITLE
[CS GO] Server update

### ISSUE THREAD

**Issue by ifrag-ru** [GameServerManagers/LinuxGSM#3129]
Good evening, today all the servers stopped starting at once, what could be the problem I can't figure out, can you help in what direction to look for the reason?

csgo-jbm@ds1492:~$ ./csgoserver start
[  OK  ] Updating csgoserver: Checking for update: SteamCMD

Update available
* Local build: 5946707
* Remote build:
* Branch: public
https://steamdb.info/app/740/

[ START ] Updating csgoserver: SteamCMD
Branch: public
Redirecting stderr to '/home/csgo-jbm/.local/share/Steam/logs/stderr.txt'
/tmp/dumps is not owned by us - delete and recreate
Unable to delete /tmp/dumps. Continuing anyway.
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...OK.

Connecting anonymously to Steam Public...Logged in OK
Waiting for user info...OK
Success! App '740' already up to date.
Complete! Updating csgoserver: SteamCMD

**Comment by DigitalFriends** [comment]
Try to change update on start in the lgsm server config, there seems to be a general problem with SteamCMD currently, i cannot update my servers still (2 days and going..) 

But you can run the server by removing "updateonstart" in the LGSM config.

**/home/csgoserver/lgsm/config-lgsm/csgoserver/csgoserver.cfg** 
```
## Updating | https://docs.linuxgsm.com/commands/update
updateonstart="**off**"
```

**Comment by ifrag-ru** [comment]
Yes really helped me at, thanks.
Tell me is this a problem only on the linuxgsm or not?
And as I understand it, now the server can be updated manually, or is it impossible to update it at all at the moment?

**Comment by rodrigo-apoc** [comment]
Same problem here

`[ FAIL ] Updating csgoserver: Checking remote build: SteamCMDcsgoserver`

`Dec 16 12:51:48.109 csgoserver: UPDATE: INFO: Using anonymous Steam login
Dec 16 12:51:49.731 csgoserver: UPDATE: PASS: Checking local build
Dec 16 12:51:58.793 csgoserver: UPDATE: FATAL: Checking remote build
Dec 16 12:51:58.799 csgoserver: UPDATE: INFO: LinuxGSM version: v20.5.1
Dec 16 12:51:58.800 csgoserver: UPDATE: FATAL: core_exit.sh exiting with code: 1`

**Comment by JoinedSenses** [comment]
Experiencing a similar issue as well on tf2 - autoupdate via `update` command failed, which is managed by cron.

**Comment by JoinedSenses** [comment]
Fix for me is to change line 45 of update_steamcmd.sh to:
`elif [ -n "${branch}" ]&&[ "${branch}" != "public" ]; then`

**Comment by includes08** [comment]
`Checking for update: SteamCMD

Update available
* Local build: 5921131
* Remote build:
* Branch: public
https://steamdb.info/app/258550/
`

same problem with rust, is this a quick fix?

**Comment by includes08** [comment]
> Fix for me is to change line 45 of home/ACCOUNT/lgsm/functions/update_steamcmd.sh to:
> `elif [ -n "${branch}" ]&&[ "${branch}" != "public" ]; then`

for whoever needs to know, hope they implement the fix soon.

**Comment by JoinedSenses** [comment]
I am still awaiting to see if my change fixed the autoupdate, would like to hear from others if they had any luck after making the change. Spent a lot of time debugging to see what was going on and i suspect either a change in the output of app_info_print or there was an oversight in the update script. Either way, the lgsm update command worked for me after making the change.

**Comment by shinySonic** [comment]
> 
> 
> I am still awaiting to see if my change fixed the autoupdate checking, would like to hear from others if they had any luck after making the change. Spent a lot of time debugging to see what was going on and i suspect either a change in the output of `app_info_print` from steamcmd or there was an oversight in the update script. Either way, the lgsm `update` command worked for me after making the change.

it sure helped me fixing the issue

**Comment by xSL0W** [comment]
> elif [ -n "${branch}" ]&&[ "${branch}" != "public" ]; then

this worked for me! thanks!
debian 10 csgo

**Comment by saheem128** [comment]
`./csgoserver v; ./csgoserver fu`
This will force the update. Worked for me!

**Comment by h3o66** [comment]
Should be fixed with the latest hotfix v20.6.2

**Comment by deathstar** [comment]
I had a similar issue on Valheim and I realized that I didn't have all the dependencies installed for LinuxGSM because I had switched server computers and reinstalled linux but copied all my data from my old server. I used `./vhserver debug` to figure this out. I presume it works similarly with csgo, you would probably do: `./csgoserver debug`.
